README.md

```diff
@@ -1,12 +1,135 @@
-# deqm-test-server
+# Data Exchange for Quality Measures (DEQM) Test Server
+
+Test server for executing FHIR-based Electronic Clinical Quality Measures (eCQMs).
+
+- [Installation](#installation)
+
+  - [Prerequisites](#prerequisites)
+  - [Local Installation](#local-installation)
+  - [Testing](#testing)
+  - [MongoDB](#mongodb)
+
+- [Usage](#usage)
+
+  - [Database Setup](#database-setup)
+  - [CRUD Operations](#crud-operations)
+  - [Searches] (#searches)
+  - [Supported FHIR Operations](#supported-fhir-operations)
+
+- [License](#license)
+
+## Installation
+
+### Prerequisites
+
+- [Node.js >=10.15.1](https://nodejs.org/en/)
+- [MongoDB >= 5.0](https://www.mongodb.com)
+- [Git](https://git-scm.com/)
+
+### Local Installation
+
+Clone the source code:
+
+```bash
+git clone https://github.com/projecttacoma/deqm-test-server.git
+```
+
+Install dependencies:
+
+```bash
+npm install
+```
+
+### Testing
+
+Unit tests can be running using the following `npm` command:
+
+```bash
+npm run test
+```
+
+### MongoDB
+
+This test server makes use of [MongoDB](https://www.mongodb.com), a cross-platform document-oriented database program.
+
+Follow the [MongoDB Community Edition installation guide](https://docs.mongodb.com/manual/installation/) for your platform, and follow the commands for running MongoDB on your machine.
+
+## Usage
+
+Once MongoDB is running on your machine, run the `npm start` command to start up the FHIR server at `localhost:3000`. The server can also be run in "watch" mode with `npm run start:watch`.
+
+For ease of testing, it is recommended to download [Insomnia API Client and Design Tool](https://insomnia.rest) for sending HTTP requests to the server and [Robo 3T](https://robomongo.org) as a GUI for viewing the Mongo database.
+
+### Database Setup
+
+The following `npm` commands can be used to set up the database:
+
+- `npm run db:setup` creates collections for all the valid FHIR resource types
+- `npm run db:delete` deletes all existing collections in the database
+- `npm run db:reset` runs both of the above, deleting all current collections and then creating new, empty collections
+
+### CRUD Operations
+
+The test server supports the standard CRUD operations to create, read, update, and delete documents in the database. See the [MongoDB CRUD Operations documentation](https://docs.mongodb.com/manual/crud/) for more information.
```

Comment:
It probably makes more sense to link to FHIR crud http page rather than mongo: https://www.hl7.org/fhir/http.html

We support create, read, update, delete, search, and transaction